### PR TITLE
Some general fixes

### PR DESCRIPTION
**Pull request by seijihariki**
Just some general fixes and an auto webui configuration checker to set img2img color correction and inpainting mask weight.

### CODE REVIEW

**Review by zero01101 — COMMENTED** [review]
i don't think i ever noticed this happening lol, and it may be placebo now but sliders do feel "better" with "out-of-bounds" mouse cursor movement... 

sigh i keep forgetting to put comments on actual code blocks -_-

File: js/ui/tool/select.js [hunk 1]
```diff
@@ -101,6 +101,10 @@ const selectTransformTool = () =>
 
 				// Selection bounding box object. Has some witchery to deal with handles.
 				const selectionBB = (x1, y1, x2, y2) => {
+					x1 = Math.round(x1);
+					y1 = Math.round(y1);
+					x2 = Math.round(x2);
+					y2 = Math.round(y2);
 					return {
```
Comment (zero01101) [inline js/ui/tool/select.js:108]: lol also don't think i ever noticed any issue here either, i almost exclusively leave grid snapping turned on

Comment (seijihariki) [reply js/ui/tool/select.js:108]: Yeah, it was having some problems when selecting things with no grid snapping:

https://user-images.githubusercontent.com/7552751/206338691-4091a3e8-9718-498c-828e-107d3a7a006d.mp4

Found another issue of no grid snapping on select.js, next commit should fix it too.

File: js/initalize/layers.populate.js [hunk 2]
```diff
@@ -59,8 +59,14 @@ mouse.registerContext(
...
-		/** @type {HTMLDivElement} */
-		const target = evn.target;
+		ctx.coords.prev.x = ctx.coords.pos.x;
+		ctx.coords.prev.y = ctx.coords.pos.y;
+
+		if (evn.layerX !== evn.clientX || evn.layerY !== evn.clientY) {
+			ctx.coords.pos.x = evn.layerX;
+			ctx.coords.pos.y = evn.layerY;
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (seijihariki) [reply js/initalize/layers.populate.js:69]: More as a performance boost for firefox, really, this one!

File: js/ui/tool/dream.js [hunk 3]
```diff
@@ -806,9 +806,13 @@ const _reticle_draw = (evn, state, tool, style = {}) => {
...
 const _dream_onwheel = (evn, state) => {
 	if (!evn.evn.ctrlKey) {
-		const v =
-			state.cursorSize -
-			Math.floor(state.config.cursorSizeScrollSpeed * evn.delta);
+		// Seems mouse wheel scroll is very different between different browsers.
+		// Will use scroll as just an event to go to the next cursor snap position instead.
+		//
+		// TODO: Someone that has a smooth scrolling mouse should verify if this works with them.
+
```
[the remark this answers is not in the document]
Comment (seijihariki) [reply js/ui/tool/dream.js:813]: Well, that's settled, then.

**Comment by seijihariki** [comment]
> i don't think i ever noticed this happening lol, and it may be placebo now but sliders do feel "better" with "out-of-bounds" mouse cursor movement...

This was because it used the current DOM layer as reference for calculating percentage. Before:

https://user-images.githubusercontent.com/7552751/206340770-82b6fb28-634a-4a28-a496-a986c56a8eb2.mp4

Now:

https://user-images.githubusercontent.com/7552751/206340889-f9b5b371-febf-436e-b96b-6f0347e20b20.mp4

**Comment by seijihariki** [comment]
I will merge the fixes for now.

**Pull request merged**